pkg/caveats/eval.go

```diff
@@ -0,0 +1,98 @@
+package caveats
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/google/cel-go/cel"
+	"github.com/google/cel-go/common/types/ref"
+	"github.com/google/cel-go/interpreter"
+	exprpb "google.golang.org/genproto/googleapis/api/expr/v1alpha1"
+)
+
+// EvaluationConfig is configuration given to an EvaluateCaveatWithConfig call.
+type EvaluationConfig struct {
+	// MaxCost is the max cost of the caveat to be executed.
+	MaxCost uint64
```

Comment:
this seems _a bit specific to CEL_. But I guess we could say "alternative caveat languages" do not support it.

Reply:
Yep, that's the idea